### PR TITLE
Alex/api/public routes

### PR DESCRIPTION
**Pull request by AlexAzalor**

### CODE REVIEW

**Review by StoliarskyiKostiantyn — APPROVED** [review]

File: api/controllers/user.py [hunk 1]
```diff
@@ -16,18 +16,20 @@
 service_alias = aliased(m.Service)
 
 
-def create_out_search_users(db_users: Sequence[m.User], lang: str, db: Session) -> list[s.UserSearchOut]:
+def create_out_search_users(
+    db_users: Sequence[m.User], lang: Literal[Language.UA, Language.EN], db: Session
```
Comment (Taraslysak) [inline api/controllers/user.py]: `lang: Language` should work?

File: api/controllers/user.py [hunk 2]
```diff
@@ -16,18 +16,20 @@
...
 
     for db_user in db_users:
         services = [
-            s.Service(uuid=service.uuid, name=service.name_ua if lang == CFG.UA else service.name_en)
+            s.Service(uuid=service.uuid, name=service.name_ua if lang.value == CFG.UA else service.name_en)
             for service in db_user.services
         ]
         regions: Result[Tuple[str, str]] = db.execute(
-            sa.select(m.Region.name_ua if lang == CFG.UA else m.Region.name_en, m.Location.uuid)
+            sa.select(m.Region.name_ua if lang.value == CFG.UA else m.Region.name_en, m.Location.uuid)
```
Comment (Taraslysak) [inline api/controllers/user.py]: `lang.value == LAnguage.UA`

File: app/models/job.py [hunk 3]
```diff
@@ -50,3 +50,18 @@ class Job(db.Model, ModelMixin):
...
+        """
+
+        jobs_count = sa.func.count(Job.id)
+        experts_count = sa.func.count(sa.func.distinct(Job.worker_id))
+
+        stmt = sa.select(Job.location_id, jobs_count, experts_count).group_by(Job.location_id)
+        result = db.execute(stmt).all()
+
+        result_dict = {row[0]: {"jobs_count": row[1], "experts_count": row[2]} for row in result}
+        return result_dict
```
Comment (Taraslysak) [inline app/models/job.py]: I think this code is a controller code, not a model class static method. It does not even use `cls` variable. Moreover, it should return a `pydantic` model, not a simple dict

File: api/routes/user.py [hunk 4]
```diff
@@ -79,3 +83,26 @@ def get_public_user_profile(
 ):
     """Returns the user profile for public view"""
     return c.get_public_user_profile(user_uuid, lang, db)
+
+
+@user_router.get("/public-top-experts/", status_code=status.HTTP_200_OK, response_model=s.PublicTopExpertsOut)
+def get_public_top_experts(
+    lang: Literal[Language.UA, Language.EN] = Language.UA,
```
Comment (Taraslysak) [inline api/routes/user.py]: lang: Language ?

File: api/routes/user.py [hunk 5]
```diff
@@ -79,3 +83,26 @@ def get_public_user_profile(
...
+    """Returns 15 best experts (for carousel)"""
+
+    has_services = m.user_services.c.user_id == m.User.id
+
+    experts = db.scalars(
+        sa.select(m.User)
+        .where(m.User.is_deleted.is_(False), has_services)
+        .order_by(m.User.average_rate.desc())
+        .distinct()
+        .limit(USER_CAROUSEL_LIMIT)
```
Comment (Taraslysak) [inline api/routes/user.py]: Put in value in config please

File: api/routes/job.py [hunk 6]
```diff
@@ -100,3 +100,13 @@ def get_jobs_on_home_page(
 ):
     """Returns jobs for home page"""
     return c.get_jobs_on_home_page(query, current_user, db)
+
+
+@job_router.get("/public-job-statistics/", status_code=status.HTTP_200_OK, response_model=s.PublicJobDict)
```
Comment (varvarashcherbyna7) [inline api/routes/job.py:106]: the name of the route does not correspond to the essence, as well

and the main route should be about jobs or job, right !
In your route description, it is indicated that these are statistics for jobs and experts
why we get users here ?

**Review by Taraslysak — COMMENTED** [review]
Please, consider some refactoring

**Pull request merged**